libs/providers/flagd/src/lib/service/grpc/service.ts

```diff
@@ -2,111 +2,231 @@ import * as grpc from '@grpc/grpc-js';
 import {
   EvaluationContext,
   FlagNotFoundError,
+  FlagValue,
   GeneralError,
   JsonValue,
   Logger,
   ParseError,
   ResolutionDetails,
+  StandardResolutionReasons,
   TypeMismatchError
 } from '@openfeature/js-sdk';
 import { GrpcTransport } from '@protobuf-ts/grpc-transport';
-import { FinishedUnaryCall, RpcError } from '@protobuf-ts/runtime-rpc';
+import type { JsonValue as PbJsonValue } from '@protobuf-ts/runtime';
+import type { UnaryCall } from '@protobuf-ts/runtime-rpc';
+import { RpcError } from '@protobuf-ts/runtime-rpc';
+import LRU from 'lru-cache';
 import { Struct } from '../../../proto/ts/google/protobuf/struct';
+import {
+  ResolveBooleanRequest,
+  ResolveBooleanResponse,
+  ResolveFloatRequest,
+  ResolveFloatResponse,
+  ResolveIntRequest,
+  ResolveIntResponse,
+  ResolveObjectRequest,
+  ResolveObjectResponse,
+  ResolveStringRequest,
+  ResolveStringResponse
+} from '../../../proto/ts/schema/v1/schema';
 import { ServiceClient } from '../../../proto/ts/schema/v1/schema.client';
-import { Config } from '../../configuration';
+import { Config, DEFAULT_MAX_CACHE_SIZE } from '../../configuration';
+import { BASE_EVENT_STREAM_RETRY_BACKOFF_MS, DEFAULT_MAX_EVENT_STREAM_RETRIES, EVENT_CONFIGURATION_CHANGE, EVENT_PROVIDER_READY } from '../../constants';
+import { FlagdProvider } from '../../flagd-provider';
 import { Service } from '../service';
 
+type AnyResponse =
+  | ResolveBooleanResponse
+  | ResolveStringResponse
+  | ResolveIntResponse
+  | ResolveFloatResponse
+  | ResolveObjectResponse;
+type AnyRequest =
+  | ResolveBooleanRequest
+  | ResolveStringRequest
+  | ResolveIntRequest
+  | ResolveFloatRequest
+  | ResolveObjectRequest;
+
+interface FlagChange {
+  type: 'delete' | 'write' | 'update';
+  source: string;
+  flagKey: string;
+}
+
+export interface FlagChangeMessage {
+  flags?: { [key: string]: FlagChange };
+}
+
 // see: https://grpc.github.io/grpc/core/md_doc_statuscodes.html
 export const Codes = {
   InvalidArgument: 'INVALID_ARGUMENT',
   NotFound: 'NOT_FOUND',
   DataLoss: 'DATA_LOSS',
-  Unavailable: 'UNAVAILABLE'
+  Unavailable: 'UNAVAILABLE',
 } as const;
 
 export class GRPCService implements Service {
-  client: ServiceClient;
+  private _client: ServiceClient;
+  private _cache: LRU<string, ResolutionDetails<FlagValue>> | undefined;
+  private _cacheEnabled = false;
+  private _streamAlive = false;
+  private _streamConnectAttempt = 0;
+  private _streamConnectBackoff = BASE_EVENT_STREAM_RETRY_BACKOFF_MS;
+  private _maxEventStreamRetries;
+  private get _cacheActive() {
+    // the cache is "active" (able to be used) if the config enabled it, AND the gRPC stream is live
+    return this._cacheEnabled && this._streamAlive;
+  }
 
-  constructor(config: Config, client?: ServiceClient) {
+  readonly streamConnection = Promise.resolve(false);
```

Comment:
@skyerus this is the equivalent "blocking" method as the one you implemented in Java.

Reply:
Can you add a comment explaining why you're calling `Promise.resolve(false)`? The context will likely be valuable to future contributors.

Reply:
Good idea, will do.